portal-api/api-tests/v1/facilities/facilities.get.api-test.js

```diff
@@ -0,0 +1,125 @@
+  ...jest.requireActual('../../../server/v1/api'),
+  getAcknowledgedAmendmentsByFacilityId: () => getAcknowledgedAmendmentsByFacilityIdMock(),
+}));
+
+describe(baseUrl, () => {
+  let maker1;
+  let mockApplication;
+  let createdFacility;
+  let testUsers;
+
+  beforeAll(async () => {
+    testUsers = await testUserCache.initialise(app);
+    maker1 = testUsers().withRole(MAKER).one();
+  });
+
+  beforeEach(async () => {
+    await databaseHelper.wipe([DB_COLLECTIONS.FACILITIES, DB_COLLECTIONS.DEALS]);
+
+    // create a GEF application/deal to attach facility to
+    mockApplication = await as(maker1).post(mockApplications[0]).to(applicationBaseUrl);
+
+    // create a facility for the deal
+    const {
+      body: { details },
+    } = await as(maker1).post({ dealId: mockApplication.body._id, type: FACILITY_TYPE.CASH, hasBeenIssued: false }).to(gefBaseUrl);
+
+    createdFacility = details;
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  withClientAuthenticationTests({
+    makeRequestWithoutAuthHeader: () => get(baseUrl),
+    makeRequestWithAuthHeader: (authHeader) => get(baseUrl, { headers: { Authorization: authHeader } }),
+  });
+
+  withRoleAuthorisationTests({
+    allowedRoles: [MAKER, CHECKER, READ_ONLY, ADMIN],
+    getUserWithRole: (role) => testUsers().withRole(role).one(),
+    makeRequestAsUser: (user) => as(user).get(baseUrl),
+    successStatusCode: 200,
+  });
+
+  it('should return the created facility', async () => {
+    // Arrange & Act
+    const { body } = await as(maker1).get(baseUrl);
+    const items = body.facilities || [];
+
+    // Assert
+    expect(items.length).toEqual(1);
+    expect(items[0]._id).toEqual(createdFacility._id);
+  });
+
+  describe('when amendments are present', () => {
+    let amendment;
+
+    beforeEach(async () => {
+      const anAcknowledgedPortalAmendment = aPortalFacilityAmendment({ status: PORTAL_AMENDMENT_STATUS.ACKNOWLEDGED });
+      // amend value and coverEndDate for the facility
+      amendment = {
+        ...anAcknowledgedPortalAmendment,
+        value: 12345,
+        facilityId: createdFacility._id,
+        coverEndDate: new Date('2030-12-31'),
+      };
+
+      jest.mocked(getAcknowledgedAmendmentsByFacilityIdMock).mockResolvedValue([amendment]);
+    });
+
+    test.each([
+      [true, 'should replace value and coverEndDate with amended values'],
+      [false, 'should not replace value and coverEndDate with amended values'],
+    ])('feature flag enabled=%s: %s', async (featureFlagEnabled) => {
+      jest.mocked(isPortalFacilityAmendmentsFeatureFlagEnabled).mockReturnValue(featureFlagEnabled);
+
+      // Arrange
+      const { body } = await as(maker1).get(baseUrl);
+      const items = body.facilities || [];
+
+      // Act
+      const responseFacilityItem = items.find((i) => `${i._id}` === `${createdFacility._id}`);
+      const response = responseFacilityItem;
+
+      // Assert
+      if (featureFlagEnabled) {
```

Comment:
Sorry @rkontogianni  we can't have if conditions in a test, we need to follow the previous approach.

* `*.ff-test.js` - FF Enabled
* `test.js` - FF Disabled

Thanks